### PR TITLE
Add ssh-agent support

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by lmakarov** [comment]
LGTM

ping @achekulaev

File: bin/dsh [hunk 1]
```diff
@@ -1468,6 +1524,10 @@ case $1 in
 	vmstat)
 		vmstat
 		;;
+	ssh-add)
```
Comment (achekulaev) [inline bin/dsh]: I don't see a need in new command. This can be done with `dsh reset ssh-agent`.

Comment (lmakarov) [inline bin/dsh:1598]: This command is for people, who use multiple ssh keys.

`dsh ssh-add -l` - list ssh identities available in ssh-agent
`dsh ssh-add -D` - remove all
`dsh ssh-add` - add the default identity (`id_rsa`)

File: bin/dsh [hunk 2]
```diff
@@ -1501,6 +1578,10 @@ case $1 in
...
+		shift
+		ssh_add $*
```
Comment (achekulaev) [inline bin/dsh]: ssh_add "$*"

File: docker-compose.yml [hunk 3]
```diff
@@ -58,6 +58,11 @@ cli:
   environment:
     # Optional. Specify a custom SSH RSA key name if necessary. Defaults to 'id_rsa'.
     - SSH_KEY_NAME=id_rsa
+    # Uncomment setting for using common ssh-agent.
```
Comment (lmakarov) [inline docker-compose.yml]: This is now baked into `blinkreaction/drupal-cli`, so not needed here.

**Comment by achekulaev** [comment]
Looks good to me

File: bin/dsh [hunk 4]
```diff
@@ -736,7 +753,48 @@ vmstat ()
...
+	if [ $? -eq 0 ] && [ "$running" == "true" ]; then
+		case "$1" in
+			"")
+				if docker exec -it ssh-agent ssh-add -l > /dev/null; then
+					return; # Some key is present
+				fi
+				if is_linux; then
+					local fullpath="${HOME}/.ssh/id_rsa"
+				else
+					local fullpath="/.home/.ssh/id_rsa"
```
Comment (achekulaev) [inline bin/dsh]: Path is incorrect

File: bin/dsh [hunk 5]
```diff
@@ -736,7 +753,48 @@ vmstat ()
...
+			-l) 
+				docker run --rm --volumes-from=ssh-agent -it blinkreaction/ssh-agent:${DRUDE_ITAG} ssh-add -l
+				exit 1
+				;;
+			*) 
+				local fullpath=$1
+				;;
+		esac
+		local directory=$(dirname "$fullpath")
+		local filename=$(basename "$fullpath")
```
Comment (achekulaev) [inline bin/dsh]: I dont see a reason to break $fullpath into pieces to only join back again 2 lines below

File: bin/dsh [hunk 6]
```diff
@@ -736,7 +753,48 @@ vmstat ()
...
+					local fullpath="/.home/.ssh/id_rsa"
+				fi
+				if [ ! -f "$fullpath" ]; then
+					echo "$fullpath: No such file or directory"
+					echo "Trying to use id_dsa"
+					local fullpath=${fullpath/id_rsa/id_dsa}
+				fi
+				;;
+			-D) 
+				docker run --rm --volumes-from=ssh-agent -it blinkreaction/ssh-agent:${DRUDE_ITAG} ssh-add -D
```
Comment (achekulaev) [inline bin/dsh]: All strings that have DRUDE_TAG should be in double quotes
`docker run --rm --volumes-from=ssh-agent -it "blinkreaction/ssh-agent:${DRUDE_ITAG}" ssh-add -D`
etc.

Please also optimize it and reuse
`local docker_run="docker run --rm --volumes-from=ssh-agent -it blinkreaction/ssh-agent:${DRUDE_ITAG}"`
and then

```
-D)
$docker_run ssh-add -D
...
-l)
$docker_run ssh-add -l
...
echo "Press ENTER or CTRL+C to skip entering passphrase."
$docker_run -v $directory:$directory ssh-add $fullpath
```

Comment (lmakarov) [inline bin/dsh]: `$docker_run -v $directory:$directory ssh-add $fullpath` is not going to work:

`Usage: docker run [OPTIONS] IMAGE [COMMAND] [ARG...]`

**Comment by achekulaev** [comment]
Looks good to me

**Pull request merged**